lua/oil/lsp/workspace.lua

```diff
@@ -76,17 +76,26 @@ local function get_matching_paths(client, filters, paths)
       ---@type string|vim.lpeg.Pattern
       local glob_to_match = glob
       if vim.glob and vim.glob.to_lpeg then
-        -- HACK around https://github.com/neovim/neovim/issues/28931
-        -- find alternations and sort them by length to try to match the longest first
-        if vim.fn.has("nvim-0.11") == 0 then
-          glob = glob:gsub("{(.*)}", function(s)
-            local pieces = vim.split(s, ",")
-            table.sort(pieces, function(a, b)
+        glob = glob:gsub("{(.*)}", function(s)
```

Comment:
We should replace the pattern with `{(.-)}` so that it's non-greedy. This is a bug with my original implementation.